patches/server/1025-Add-configuration-option-for-MiniMessage-formatted-M.patch

```diff
@@ -0,0 +1,38 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: wordandahalf <10679776+wordandahalf@users.noreply.github.com>
+Date: Thu, 4 Aug 2022 22:09:28 -0500
+Subject: [PATCH] Add configuration option for MiniMessage-formatted MOTD
+
+
+diff --git a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+index 456595e4b7e0c7f50617aa2694b0d2dfc368ab81..446ed729a74ba08d882059980652b29cf3a872cd 100644
+--- a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
++++ b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+@@ -45,6 +45,8 @@ public class GlobalConfiguration extends ConfigurationPart {
+         }
+ 
+         public Component noPermission = Component.text("I'm sorry, but you do not have permission to perform this command. Please contact the server administrators if you believe that this is in error.", NamedTextColor.RED);
++        @Nullable
++        public Component motd = null;
```

Comment:
im not sure how is handled the null in the file... and what can happen if you use a empty string in the config..

Reply:
From my testing, the config field is omitted by default. If added, its value overrides that of `motd` in `server.properties` so long as it is not empty:
```yaml
# Is removed, considered null
motd:
# ""
motd: null
# Considered a value, yeilds an empty MOTD
motd: ''
```
Would it be preferred to have the field an empty string by default, and have it override only when it is nonempty?

Reply:
I generally eer towards using a string like "default", as then we can just document the behavior of what's going on, vs empty string which, what if you do want an empty string? In the future there is an argument to move all of the config strings over into paper.yml, and I'm not 100% sure that in the long term we'll want to keep those options existing in their standard config files (outside of reading them if they exist so that we can convert them). "default" just offers a sane option which allows us to determine and change what the default behavior is in the future without worrying as much about the side-effects of it (default right now means grab from server.properties, in the future, it could be, use the default hard-coded string)

Reply:
I see your perspective, but making it nullable seems to be as the simplest and most straightfoward approach. It's completely opt-in, otherwise it functions exactly as it did before. And it's pretty much self-documenting. You want a MiniMessage-formatted MOTD? Add the field. You want to use the value in `server.propertes`? Remove the field.